AK/Handle.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022, Tim Schumacher <timschumi@gmx.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/NonnullOwnPtr.h>
+#include <AK/Variant.h>
+
+namespace AK {
+
+template<typename T>
+class Handle {
```

Comment:
What a funny way to spell `MaybeOwned<T>` /s

Reply:
joke aside though, this name doesn't really make sense when it's not in ns `Core::Stream`, now it just looks like a typed integer replacement that you'd vend from random containers, not a "handle to a stream".

Reply:
> now it just looks like a typed integer replacement that you'd vend from random containers

I'm afraid I don't see it, but maybe I haven't worked with other container implementations for long enough. Are you talking about something like "this is a reference/index to that element inside a container"?

> [...], not a "handle to a stream"

That was kind of the intention behind removing the Stream-derivative restriction, since I imagined that - once `Handle` is in AK - other parts of the code might want to make use of it too. Maybe it became a bit too generic in the process.

> What a funny way to spell `MaybeOwned<T>` /s

I'd not actually be opposed to renaming it to that, apart from the usual "mah conflicts!".

Reply:
> Are you talking about something like "this is a reference/index to that element inside a container"?

nah, a handle is the usual term for an integer that's interpreted by the container and/or whatever you're interfacing...at least as far as I know. `MaybeOwned` is fine though, it's a bit of a meme name, but can't go wrong with just enough meme.